Facebook login

ivangarzab opened the pull request.

ivangarzab committed:
Implement Facebook Login Feature & add Butterknife:
-  Add Facebook SDK
-  Create LoginButton in activity_login
-  Update Manifest with Facebook Activities
-  Add Butterknife to Project

ivangarzab committed:
Add google-service.json file

itzcoatl90 commented on app/src/main/java/mcs/salazar/jesus/mcsbattleship/LoginActivity.java:
```diff
@@ -1,13 +1,114 @@
+        // Initialize Facebook Login button
+        mCallbackManager = CallbackManager.Factory.create();
+        // Defined the information needed from user
+        mFacebookButton.setReadPermissions("email");
```
Can we make "email" to be in strings.xml and call it from there?

itzcoatl90 commented on app/src/main/java/mcs/salazar/jesus/mcsbattleship/LoginActivity.java:
```diff
@@ -1,13 +1,114 @@
+        mFacebookButton.registerCallback(mCallbackManager, new FacebookCallback<LoginResult>() {
+            @Override
+            public void onSuccess(LoginResult loginResult) {
+                Log.d("TRASH", "facebook:onSuccess:" + loginResult);
```
Can we change "TRASH" to the name of the class and have the name of the class in a private static final String TAG?

itzcoatl90 replied:
Actually public so it can be accessed from the callback

itzcoatl90 replied:
Also, I know is will going a P.I.N.A. but can we also pass to string.xml also "facebook:onSuccess:"? I know I know... but the code is cleaner if it doesn't have hardcoded values, and if you are used to do it, you will avoid any risk having issues with proguard

itzcoatl90 commented on app/src/main/java/mcs/salazar/jesus/mcsbattleship/LoginActivity.java:
```diff
@@ -1,13 +1,114 @@
+
+            @Override
+            public void onCancel() {
+                Log.d("TRASH", "facebook:onCancel");
```
Same here

itzcoatl90 replied:
String.xml

itzcoatl90 commented on app/src/main/java/mcs/salazar/jesus/mcsbattleship/LoginActivity.java:
```diff
@@ -1,13 +1,114 @@
+
+            @Override
+            public void onError(FacebookException error) {
+                Log.d("TRASH", "facebook:onError", error);
```
Same

itzcoatl90 replied:
String.xml

itzcoatl90 commented on app/src/main/java/mcs/salazar/jesus/mcsbattleship/LoginActivity.java:
```diff
@@ -1,13 +1,114 @@
+    }
+
+    private void handleFacebookAccessToken(AccessToken token) {
+        Log.d("TRASH", "handleFacebookAccessToken:" + token);
```
TAG

itzcoatl90 replied:
String.xml

itzcoatl90 commented on app/src/main/java/mcs/salazar/jesus/mcsbattleship/LoginActivity.java:
```diff
@@ -1,13 +1,114 @@
+                    public void onComplete(@NonNull Task<AuthResult> task) {
+                        if (task.isSuccessful()) {
+                            // Sign in success, update UI with the signed-in user's information
+                            Log.d("TRASH", "signInWithCredential:success");
```
TAG

itzcoatl90 replied:
String.xml

itzcoatl90 commented on app/src/main/java/mcs/salazar/jesus/mcsbattleship/LoginActivity.java:
```diff
@@ -1,13 +1,114 @@
+                            //FirebaseUser user = mAuth.getCurrentUser();
+                        } else {
+                            // If sign in fails, display a message to the user.
+                            Log.w("TRASH", "signInWithCredential:failure",
```
TAG

itzcoatl90 commented on app/src/main/java/mcs/salazar/jesus/mcsbattleship/LoginActivity.java:
```diff
@@ -1,13 +1,114 @@
+                            Log.w("TRASH", "signInWithCredential:failure",
+                                    task.getException());
+                            Toast.makeText(LoginActivity.this,
+                                    "Authentication failed! Please, try again",
```
String.xml .   It's not a good thing to hardcode thinks... I know, I know... please

itzcoatl90 commented on app/src/main/res/layout/activity_login.xml:
```diff
@@ -6,10 +6,14 @@
+        android:id="@+id/login_facebook_button"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:text="Hello World!"
```
This one also should be in String.xml because of locales changes... what happens if we want to translate to spanish?

ivangarzab replied:
That line was deleted from the default template

itzcoatl90 commented on app/src/main/res/layout/activity_login.xml:
```diff
@@ -6,10 +6,14 @@
         android:layout_height="wrap_content"
-        android:text="Hello World!"
+        android:layout_gravity="center_horizontal"
+        android:layout_marginTop="30dp"
```
This measures, can we have them in a dimens.xml?        It would be more maneagable

ivangb6 replied:
I'd love an example of dimens.xml, I've never done that

itzcoatl90 commented on app/src/main/res/layout/activity_login.xml:
```diff
@@ -6,10 +6,14 @@
-        android:text="Hello World!"
+        android:layout_gravity="center_horizontal"
+        android:layout_marginTop="30dp"
+        android:layout_marginBottom="30dp"
```
same

itzcoatl90 commented on app/src/main/res/values/strings.xml:
```diff
@@ -1,3 +1,5 @@
 <resources>
     <string name="app_name">MCSBattleShip</string>
+    <string name="facebook_app_id">147216785969353</string>
+    <string name="fb_login_protocol_scheme">fb147216785969353</string>
```
This file should be bigger... lol .    YO HO HO!

ivangarzab replied:
Have some rum, captain!

bereket11 reviewed (commented):
I use "firebaseAuth" as variable name for Firebase object.

ivangarzab committed:
Fix comment issues for Pull Request
-  Add all strings to strings.xml
-  Create dimens.xml
-  Globalize the TAG for Logging messages

bereket11 committed:
Adding login using email and password

bereket11 committed:
Respond PR and commented to it

ivangarzab committed:
Merge pull request #2 from ivangb6/email_login

Adding login using email and password
